src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java

```diff
@@ -472,12 +473,12 @@ public ItemStack removeStackFromSlot(final int index)
         if (nonnulllist != null && !(nonnulllist.get(tempIndex)).isEmpty())
```

Comment:
Here is still an isEmpty() check.
Please replace that with the ItemStackUtil one.